CitationStyleLanguagePlugin.php

```diff
@@ -583,15 +583,23 @@ public function getCitation(PKPRequest $request, Submission $submission, string
                     $additionalMarkup = [
                         'DOI' => [
                             'function' => function ($item, $renderedValue) {
-                                return '<a href="https://doi.org/' . $item->DOI . '">' . $renderedValue . '</a>';
+                                $prefixPattern = '/.*DOI:/i';
```

Comment:
This is a regular expression to match DOIs: `/^10.\d{4,9}/[-._;()/:A-Z0-9]+$/i` (taken from https://www.crossref.org/blog/dois-and-matching-regular-expressions/), notice that it accepts characters and also `:`, so there's room for failures here.

Reply:
What's possible to come from the `$renderedValue`? Isn't it possible to use something like `/^DOI:/i` or `/^\s*DOI:/i`?